### PR TITLE
Feat: 🚀 Prepared addon for Kube-State-Matrics

### PR DESCRIPTION
**Pull request by nileshgadgi**
## what
* Feature Kube-State-Metrics addon.
* Remove unnecessary vars.

## why
* Added kube-state-metrics component to expose cluster-level metrics for monitoring and alerting

## references
* https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-state-metrics
* https://artifacthub.io/packages/helm/prometheus-community/kube-state-metrics

### CODE REVIEW

File: _examples/complete/config/override-kube-state-matrics.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: "eks.amazonaws.com/nodegroup"
+          operator: In
+          values:
+          - "test1-eks-test1-cluster-critical-nodes"
```
Comment (h1manshu98) [inline _examples/complete/config/override-kube-state-matrics.yaml:11]: Fix EKS NodeGroup name

File: addons/kube-state-metrics/data.tf
Comment (h1manshu98) [inline addons/kube-state-metrics/data.tf:1]: Remove empty files if not required

File: addons/kube-state-metrics/locals.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    dependency_update          = try(var.kube_state_metrics_extra_configs.dependency_update, "false")
+    replace                    = try(var.kube_state_metrics_extra_configs.replace, "false")
+  }
+
+  kube_state_metrics_extra_configs = var.kube_state_metrics_extra_configs
+
+  helm_config = merge(
+    local.default_helm_config,
+    var.helm_config,
+    local.kube_state_metrics_extra_configs
```
Comment (h1manshu98) [inline addons/kube-state-metrics/locals.tf:42]: can we remove `local.kube_state_metrics_extra_configs` from inside `helm_config` attribute?

File: addons/kube-state-metrics/variables.tf [hunk 3]
```diff
@@ -0,0 +1,32 @@
+variable "helm_config" {
+  description = "Helm provider config for Metrics Server"
```
Comment (h1manshu98) [inline addons/kube-state-metrics/variables.tf:2]: Fix description

File: addons/velero/locals.tf [hunk 4]
```diff
@@ -34,8 +34,6 @@ locals {
     replace                    = try(var.velero_extra_configs.replace, "false")
   }
 
-  velero_extra_configs = var.velero_extra_configs
```
Comment (h1manshu98) [inline addons/velero/locals.tf:37]: `velero_extra_configs` attribute can not be removed, this is required.

File: addons/kube-state-metrics/locals.tf (hunk 2 above)
Comment (nileshgadgi) [reply addons/kube-state-metrics/locals.tf:42]: You sure that with that we can pass extra argument if we want?

File: addons/velero/locals.tf (hunk 4 above)
Comment (nileshgadgi) [reply addons/velero/locals.tf:37]: this was not used anywhere, as terraform cleaner tool.

Comment (h1manshu98) [reply addons/velero/locals.tf:37]: FYR : https://github.com/clouddrove/terraform-aws-eks-addons/blob/master/_examples/complete/variables.tf#L120-L127

**Review by themaniskshah — APPROVED** [review]
LGTM!

**Review by vaibhav7797 — APPROVED** [review]
LGTM!

**Review by anmolnagpal — APPROVED** [review]

**Pull request merged**